draft-tuexen-tsvwg-rfc4960-errata.xml

```diff
@@ -733,6 +733,79 @@ partial_bytes_acked and cwnd in the congestion avoidance phase.</t>
 </section>
 </section>
 
+<section title='HEARTBEAT ACK and the association error counter'>
+<section title='Description of the Problem'>
+<t>Section 8.1 and Section 8.3 of <xref target="RFC4960"/>
+prescribes that the receiver of a HEARTBEAT ACK must reset the
+association overall error counter. In some circumstances, e.g. when
+a router discards DATA chunks but not HEARTBEAT chunks due to the
+larger size of the DATA chunk, it is better to not clear the
+association error counter by the HEARTBEAT ACK and reset it by the
+SACK only.</t>
+</section>
+<section title='Text Changes to the Document'>
+<figure>
+<artwork>
+---------
+Old text: (Section 8.1)
+---------
+
+The counter shall be reset each time a DATA chunk sent to that peer
+endpoint is acknowledged (by the reception of a SACK) or a HEARTBEAT
+ACK is received from the peer endpoint.
+
+---------
+New text: (Section 8.1)
+---------
+
+The counter shall be reset each time a DATA chunk sent to that peer
+endpoint is acknowledged (by the reception of a SACK). When a
+HEARTBEAT ACK is received from the peer endpoint, the counter should
+also be reset. The receiver of the HEARTBEAT ACK may choose not to
+clear the counter while transmitting data on the association. Such
+choice may be motivated by different sizes of the DATA and HEARTBEAT
+chunks. 
+
+---------
+Old text: (Section 8.3)
+---------
+
+Upon the receipt of the HEARTBEAT ACK, the sender of the HEARTBEAT
+should clear the error counter of the destination transport address
+to which the HEARTBEAT was sent, and mark the destination transport
+address as active if it is not so marked.  The endpoint may
+optionally report to the upper layer when an inactive destination
+address is marked as active due to the reception of the latest
+HEARTBEAT ACK.  The receiver of the HEARTBEAT ACK must also clear the
+association overall error count as well (as defined in Section 8.1).
+
+---------
+New text: (Section 8.3)
+---------
+
+Upon the receipt of the HEARTBEAT ACK, the sender of the HEARTBEAT
+should clear the error counter of the destination transport address
+to which the HEARTBEAT was sent, and mark the destination transport
+address as active if it is not so marked. The receiver of the
+HEARTBEAT ACK may choose not to clear the error counter of the
+destination transport address if there are valid reasons for it, e.g.
+outstanding data on the destination address. The endpoint may
+optionally report to the upper layer when an inactive destination
```

Comment:
I didn't plan to add more details. I tried to align the descriptions of the destination error counter reset and the association error counter reset. 
Would you accept one of the following proposals: 
"The receiver of the HEARTBEAT ACK may choose not to clear the error counter of the destination transport address if there are valid reasons for it."
Or
"The receiver of the HEARTBEAT ACK may choose not to clear the error counter of the destination transport address if there is outstanding data to the destination address."